src/commonMain/kotlin/fr/acinq/lightning/serialization/v4/Serialization.kt

```diff
@@ -144,6 +145,11 @@ object Serialization {
                 write(0x00)
             }
         }
+        if (liquidityPurchases.isNotEmpty()) {
+            write(0x01)
+            writeNumber(liquidityPurchases.size)
+            liquidityPurchases.forEach { writeLiquidityPurchase(it) }
```

Comment:
```suggestion
            writeCollection(liquidityPurchases) { writeLiquidityPurchase(it) }
```

Reply:
On 2nd thought I think we should also remove the `if`, and always write the `0x01` and the leases, even if the list is empty. Otherwise, if we want to add a 2nd optional field later we will have to do the same kind of tricks as for `liquidityPurchase/localCommit` to handle cases where there the middle object is not defined (instead of simply reading objects one after the other).

Reply:
Done in https://github.com/ACINQ/lightning-kmp/pull/561/commits/f735f45eee0b765ed04563ecfad0f5a0b061a774